packages/vm/src/calls.rs

```diff
@@ -39,58 +40,76 @@ pub fn call_query<S: Storage + 'static, A: Api + 'static>(
     msg: &[u8],
 ) -> Result<Result<QueryResponse, ApiError>, Error> {
     let data = call_query_raw(instance, msg)?;
-    let res: QueryResult = from_slice(&data)?;
-    Ok(res.into())
+    let api_result: QueryResult = from_slice(&data)?;
+    let result: Result<QueryResponse, ApiError> = api_result.into();
+
+    // Ensure query response is valid JSON
+    if let Ok(binary_response) = &result {
+        serde_json::from_slice::<serde_json::Value>(binary_response.as_slice()).or(RuntimeErr {
```

Comment:
Ah, here you enforce the results in the VM. Some comments:

1. I wonder about using serde_json - last time I tried, it brought in floating point ops, which we reject in bytecode. Can you ensure this doesn't happen.
2. It seems `serde::de::IgnoredAny` is a better type for this: https://users.rust-lang.org/t/serde-json-checking-syntax-of-json-file/16265/4 since we don't use the result. Maybe that would also work with serde-json-wasm (or maybe not?)
3. Do we want to add checks on the contract (client side)? 
4. Are there other operations we can optimize here assuming json? Or just on the go/client side?

Reply:
1. The dependency is added to the VM only, not the standard library. Thus this only affects the host and is never compiled to wasm.
2. I found this https://github.com/CosmWasm/serde-json-wasm/blob/master/src/de/mod.rs#L339-L340 which makes me think this does not work with serde-json-wasm
3. The contract is responsible for returning data. If it decides to return trash, integration tests will detect that. If there is a way to model _any valid JSON_ in the contract's type system, that's a little help for the developer at least. But even without that, templates and examples should do the trick.
4. The go side does not even run this check. Since ot calls `call_query_raw` directly, the check is bypassed (as well as all the other type checks). But this problem exists independent of this change. What we really optimize is what clients need to expect. They now can require valid JSON or blame the contract and don't need to adapt to every possible binary result.

Reply:
1. Good point.
2. Thanks for checking
3. Good to check this in the VM side (enforce it close to the contract). I will make another PR to make it harder for devs to even mess up.

Looks good